client/my-sites/email/email-forwards-add/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { Card } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import page from 'page';
+import { useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import DocumentHead from 'calypso/components/data/document-head';
+import QueryProductsList from 'calypso/components/data/query-products-list';
+import QuerySiteDomains from 'calypso/components/data/query-site-domains';
+import HeaderCake from 'calypso/components/header-cake';
+import Main from 'calypso/components/main';
+import SectionHeader from 'calypso/components/section-header';
+import PageViewTracker from 'calypso/lib/analytics/page-view-tracker';
+import EmailForwardingAddNewCompactList from 'calypso/my-sites/email/email-forwarding/email-forwarding-add-new-compact-list';
+import EmailHeader from 'calypso/my-sites/email/email-header';
+import AddEmailAddressesCardPlaceholder from 'calypso/my-sites/email/gsuite-add-users/add-users-placeholder';
+import {
+	emailManagement,
+	emailManagementAddEmailForwards,
+	emailManagementPurchaseNewEmailAccount,
+} from 'calypso/my-sites/email/paths';
+import getCurrentRoute from 'calypso/state/selectors/get-current-route';
+import {
+	hasLoadedSiteDomains,
+	isRequestingSiteDomains,
+} from 'calypso/state/sites/domains/selectors';
+import { getSelectedSite } from 'calypso/state/ui/selectors';
+
+import './style.scss';
+
+type EmailForwardsAddProps = {
+	selectedDomainName: string;
+	source?: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-empty-function
+const noop = (): void => {};
+
+const EmailForwardsAdd = ( { selectedDomainName, source }: EmailForwardsAddProps ): JSX.Element => {
+	const currentRoute = useSelector( getCurrentRoute );
+	const selectedSite = useSelector( getSelectedSite );
+
+	const hasLoadedDomains = useSelector( ( state ) =>
+		hasLoadedSiteDomains( state, selectedSite?.ID ?? null )
+	);
+	const requestingSiteDomains = useSelector( ( state ) =>
+		Boolean( selectedSite && isRequestingSiteDomains( state, selectedSite.ID ) )
+	);
+	const areDomainsLoading = requestingSiteDomains || ! hasLoadedDomains;
```

Comment:
This could potentially be misleading if an error occurs, and site domains haven't loaded yet.

Reply:
I don't think that should occur: `requestingSiteDomains` should only be true while a request is in flight, and `hasLoadedDomains` should be true once the request completes, even if the request fails. (It should also be noted that `hasLoadedDomains` may be true all along if we have previously loaded the domains.)